h/static/scripts/discovery.coffee

```diff
@@ -86,6 +86,15 @@ module.exports = class Discovery
   _onMessage: (event) =>
     {source, origin, data} = event
 
+    # If `origin` is 'null' the source frame is a file URL or loaded over some
+    # other scheme for which the `origin` is undefined. In this case, the only
+    # way to ensure the message arrives is to use the wildcard origin. See:
+    #
+    #   https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
```

Comment:
Firefox and Chrome exhibit the same behavior at looking at the implementation, it looks like they are referencing this spec: https://hyp.is/AVQ4XdxnH9ZO4OKSl1Db/tools.ietf.org/html/rfc6454